src/lib/mina.ts

```diff
@@ -266,10 +271,33 @@ function LocalBlockchain({
     },
     sendTransaction(txn: Transaction) {
       txn.sign();
+
+      let partiesJson = partiesToJson(txn.transaction);
+
       ledger.applyJsonTransaction(
-        JSON.stringify(partiesToJson(txn.transaction)),
+        JSON.stringify(partiesJson),
         String(accountCreationFee)
       );
+
+      // fetches all events from the transaction and stores them
+      // events are identified and associated with a publicKey and tokenId
+      partiesJson.otherParties.forEach((p: any) => {
```

Comment:
nit: I believe the `any` annotation in the callback unnecessarily removes the type information on `p`. typescript should now that `p: Party` without a type annotation